### PR TITLE
add mongodb app in integration test

### PR DESCRIPTION
**Pull request by viveksinghggits**
## Change Overview

- This change adds mongodb apps integration into our testing framework. There are some intermittent issues, that I will try and fix as soon as possible. 

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan
Ran go test command to test this change.
```
go test -v ./pkg/testing -tags=integration -timeout 30m -count=1
```

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/app/mongodb.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+		Namespace: mongo.namespace,
+	}
+}
+
+func (mongo *MongoDB) Uninstall(ctx context.Context) error {
+	cli := helm.NewCliClient(helm.V3)
+
+	log.Print("Uninstalling application.", field.M{"app": mongo.name})
+	err := cli.Uninstall(ctx, mongo.name, mongo.namespace)
+	if err != nil {
```
Comment (PrasadG193) [inline pkg/app/mongodb.go]: We can just return `errors.Wrapf(err)`here. If error is nil errors.Wrapf() also return nil

File: pkg/app/mongodb.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		return errors.Wrapf(err, "Error %s checking if the pod is master.", stderr)
+	}
+
+	// extract the ismaster field from the output and make sure
+	// this ismaster field is true
+	re := regexp.MustCompile("\"ismaster\" : ([a-z]*),")
+	match := re.FindStringSubmatch(isMasterStdout)
+
+	isMaster, err := strconv.ParseBool(match[1])
```
Comment (PrasadG193) [inline pkg/app/mongodb.go]: We should ensure `len(match) >= 2` before accessing the element

File: pkg/app/mongodb.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+}
+func (mongo *MongoDB) Reset(ctx context.Context) error {
+	log.Print("Resetting the application.", field.M{"app": mongo.name})
+	// delete all the entries from the restaurants dummy colleciton
+	// we are not deleting the database becasue we are dealing with admin database here
+	// and deletion admin database is prohibited
+	deleteDBCMD := []string{"sh", "-c", fmt.Sprintf("mongo admin --authenticationDatabase admin -u %s -p $MONGODB_ROOT_PASSWORD --quiet --eval \"rs.slaveOk(); db.restaurants.drop()\"", mongo.username)}
+	stdout, stderr, err := mongo.execCommand(ctx, deleteDBCMD)
+	if err != nil {
+		return errors.Wrapf(err, "Error %s, resetting the mongodb application. stdout is %s", stderr, stdout)
```
Comment (PrasadG193) [inline pkg/app/mongodb.go]: No need to check for err here. We can just return `errors.Wrapf`

File: pkg/app/mongodb.go (hunk 1 above)
Comment (viveksinghggits) [reply pkg/app/mongodb.go]: yeah, I understand, but if there is an error I wanted to print when did we encounter the error. Do you think `errors.Wrapf(err)` is better idea.

File: pkg/app/mongodb.go (hunk 2 above)
Comment (viveksinghggits) [reply pkg/app/mongodb.go]: So, output of `db.ismaster` is always going to have `ismaster` field in output so we are always going to have `len(match)>= 2`. But yeah, I think we should that.

File: pkg/app/mongodb.go (hunk 1 above)
Comment (PrasadG193) [reply pkg/app/mongodb.go]: We are anyways returning `errors.Wrapf(err, "Error while uninstalling the application.")` on line no 134. What I meant is we don't need to check `if err != nil` if we are using `errors.Wrapf()` function

File: pkg/app/mongodb.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+	"github.com/kanisterio/kanister/pkg/helm"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/log"
+)
+
+type MongoDB struct {
+	cli       kubernetes.Interface
+	namespace string
+	username  string
+	password  string
```
Comment (PrasadG193) [inline pkg/app/mongodb.go]: @viveksinghggits are we using this field?

**Pull request merged**